azure-pipelines/end-to-end-tests-dir/asset-caching.ps1

```diff
@@ -1,6 +1,119 @@
+$actual = $actual -replace "`r`n", "`n"
+
+$expected = @(
+"A suitable version of .* was not found \(required v[0-9\.]+\)."
+"Downloading .*."
+) -join "`n"
+
+if (-not ($actual -match $expected)) {
+    throw "Failure: asset cache miss (not configured) + x-block-origin disabled"
+}
+
+# Testing asset cache miss (configured) + x-block-origin enabled
+Refresh-TestRoot
+$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite;x-block-origin", "--downloads-root=$DownloadsRoot"))
+$actual = $actual -replace "`r`n", "`n"
+
+$expected = @(
+"A suitable version of .* was not found \(required v[0-9\.]+\)."
+"Asset cache miss for .* and external downloads are blocked by x-block-origin"
+) -join "`n"
+
+if (-not ($actual -match $expected)) {
+    throw "Failure: asset cache miss (configured) + x-block-origin disabled"
+}
+
+# Testing asset cache miss (configured) + x-block-origin disabled
+Refresh-TestRoot
+$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite;", "--downloads-root=$DownloadsRoot"))
+$actual = $actual -replace "`r`n", "`n"
+
+$expected = @(
+"A suitable version of .* was not found \(required v[0-9\.]+\)."
+"Asset cache miss; downloading from .*"
+"Successfully stored .* to mirror."
+) -join "`n"
+
+if (-not ($actual -match $expected)) {
+    throw "Failure: asset cache miss (configured) + x-block-origin disabled"
+}
+
+# Testing asset cache hit
+Refresh-Downloads
+Run-Vcpkg -TestArgs ($commonArgs + @('remove', 'vcpkg-internal-e2e-test-port'))
+$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite;", "--downloads-root=$DownloadsRoot"))
+$actual = $actual -replace "`r`n", "`n"
+
+$expected = @(
+"A suitable version of .* was not found \(required v[0-9\.]+\)."
+"Asset cache hit for .*."
+) -join "`n"
+
+if (-not ($actual -match $expected)) {
+    throw "Failure: asset cache hit"
+}
+
+# Testing asset cache hit with --debug
+Refresh-Downloads
+Run-Vcpkg -TestArgs ($commonArgs + @('remove', 'vcpkg-internal-e2e-test-port'))
+$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite;", "--downloads-root=$DownloadsRoot", "--debug"))
+$actual = $actual -replace "`r`n", "`n"
+
+$expectedPattern = "(?s)" + ".*\[DEBUG\] Downloaded from: .*"
```

Comment:
I don't think debug output should be 'salient'. The intent of debug output is to help us debug the tool when something horrible goes wrong, and any scenario that requires a customer to say debug is bad.

In particular, debug output is not localized.

Reply:
I added a note to talk about this with the other maintainers on Thursday.